config/cache/v1alpha1/cache.proto

```diff
@@ -63,6 +63,48 @@ message ResourceQuantity {
   string cpu = 2;
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a data source connection. A map is available for passing implementation specific
+// properties.
+message DataSourceSpec {
+  // +kubebuilder:validation:Enum=POSTGRES_14;MYSQL_8;SQL_SERVER_2019
+  // Type and version of the underlaying DB. Needed to decide which drivers need to be used
+  optional DBType db_type = 1;
+  // Additional properties. DB specific
+  map<string, string> connection_properties = 2;
+  // Reference to a local secret containing DB connection details.
+  LocalObjectRef secret_ref = 3;
+  // Reference to ServiceBinding provider
+  ServiceRef service_provider_ref = 4;
+}
+
+// LocalObjectRef contains enough information to let you locate the referenced object inside the same namespace.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+message LocalObjectRef {
```

Comment:
Maybe `LocalRef` is more consistent with the existing `NamespacedRef`?

Reply:
I was trying to emulate the k8s core api here, but that should be `LocalObjectReference`.

We can do:
1. `LocalObjectReference` and `NamespacedObjectReference`
2. `LocalRef` and `NamespacedRef`

WDYT?

Reply:
I would say 2 just to minimized the changes, but 1 it's actually better. Let's do 1 and see how it's easy to propagate an api change :)

Reply:
API and Operator PR updated. There wasn't too many changes required luckily :slightly_smiling_face: